Backend/CourseService.cs

```diff
@@ -58,13 +58,13 @@ public List<Course> FindByStartName(string courseStartsWith)
         /// If course with "courseName" exists function return ID, if course doesnt exist fuction
         /// save this course and return id of saved course.
         /// </summary>
-        public async Task<Course> GetOrAddNotExistsCourseByShortcut(string courseShortcut)
+        public async Task<Course> GetOrAddNotExistsCourseByShortcut(string courseShortcut, string courseName = null)
```

Comment:
What is the difference between this method and ..ByName..
Now it look the same, if yes, you should refactro it (delete one and rename second (without Shortcut or name)..

Reply:
ByName searches by CourseName, ByShortcut searches by CourseShortcut. I added optional parameter course because if it does not find subject, it inserts it (If it was'nt  there, subject would be inserted without CourseName).

Reply:
OK, but these two methods are almost the same (except first line - I miss this before). So at least I recommend to extract the whole code from these two methods (except first line) to separate private helper method and call it from both methods (lines 64-83, and 89- 108). I know it is not your code, but it is good time to refactor it a little :-)